runtime/dart_controller.cc

```diff
@@ -146,11 +153,29 @@ tonic::DartErrorHandleType DartController::RunFromSource(
 void DartController::CreateIsolateFor(const std::string& script_uri,
                                       const uint8_t* isolate_snapshot_data,
                                       const uint8_t* isolate_snapshot_instr,
+                                      const std::vector<uint8_t>& platform_kernel,
                                       std::unique_ptr<UIDartState> state) {
   char* error = nullptr;
-  Dart_Isolate isolate = Dart_CreateIsolate(
-      script_uri.c_str(), "main", isolate_snapshot_data, isolate_snapshot_instr,
-      nullptr, static_cast<tonic::DartState*>(state.get()), &error);
+
+  Dart_Isolate isolate;
+  if (!platform_kernel.empty()) {
+    // Copy kernel bytes so they won't go away after we exit this method.
+    // This is needed because original kernel data has to be available
+    // during code execution.
+    uint8_t* platform_kernel_data = (uint8_t*) malloc(platform_kernel.size());
+    memcpy(platform_kernel_data, platform_kernel.data(), platform_kernel.size());
+
```

Comment:
Agree, I was wondering about the multiple copies being created for each isolate until I realized the service isolate is still being started the old way. If we switched the service isolate to also use the kernel platform file then you may have to save this once and not keep re-reading this everytime.